docs/pages/mydoc/crdts.md

```diff
@@ -0,0 +1,240 @@
+---
+
+
+Most highly available databases only provide last-writer-wins registers, which resolve concurrent updates using timestamps.
+Antidote offers a variety of other datatypes with smarter strategies to resolve concurrent updates.
+The available data types are described below.
+
+All data types have in common, that their value only depends on the operations which have been delivered to the currently used Antidote data center.
+For all datatypes there can be concurrent operations which are not yet visible.
+
+In the descriptions below the word "after" always means "causally after".
+An operation which happens later in time can be concurrent, if it happened independently.
+
+Many datatypes support a `reset` operation, which is used when nesting a datatype in a map-datatype.
+Therefore the details of `reset` are explained with the map-datatypes below.
+
+
+
+## Last-writer-wins Register (LWW-Register)
+
+Reading a Last-writer-wins Register (LWW-Register) returns the latest written value based on timestamps.
+The timestamps respect causality, so assigning a new value is guaranteed to overwrite previous values.
+
+The initial value is the empty binary.
+
+#### Operations
+
+assign(value)
+:	Assigns a value to the register.
+
+## Multi-value Register (MV-Register)
+
+Reading a Multi-value Register (MV-Register) returns a list of all concurrently assigned values, which have not been overridden by other assignments.
+The returned list is ordered by the lexicographic order on the byte representation of the value.
+
+In the initial state reading the register returns the empty list.
+
+
+#### Operations
+
+assign(value)
+:	Assigns a value to the register.
+
+reset()
+:   Resets the register to the initial state (empty list).
+
+## Number Types
+
+### Counter
+
+The counter datatype only allows incrementing and decrementing the value.
+Reading the value returns the aggregated value from all operations.
+
+#### Operations
+
+increment(v)
+:	Increments the counter by `v`.
+
+decrement(v)
+:	Decrements the counter by `v`.
+
+reset()
+:   Resets the register to the initial state (empty list).
+
+
+### Fat Counter
+
+The fat counter offers the same functionality as the counter above but additionally supports the `reset` operation, which is useful for embedding the counter in a map datatype.
+
+The cost of this additional feature is an increased storage size, which is linear in the number of operations since the last reset.
+
+
+#### Operations
+
+increment(v)
+:	Increments the counter by `v`.
+
+decrement(v)
+:	Decrements the counter by `v`.
+
+reset()
+:   Resets the counter to the initial state (0).
+
+
+
+### Integer
+
+The integer extends the counter with an additional `set`-operation which allows to assign a value to the integer.
+
+Intuitively concurrent updates are resolved by taking the maximum in the case of concurrent `set`-operations and by considering all increments and decrement operations which are not overwritten by the `set`-operation.
```

Comment:
Just to understand...  if we start with a counter with value 1 and concurrently we do:

rep1: inc, inc , inc, set(2)

rep2: set(2)

we end up with 5, right? This does not make sense to me.